### ISSUE TITLE
False positive for version number with leading zero

### ISSUE THREAD

**Issue by MaxTranced** [stackrox/stackrox#5278]
```
certifi - 2022.12.7
CVE-2022-23491
CVSS Score: 7.5
Fixed by version: 2022.12.07
CVE Link: [here](https://nvd.nist.gov/vuln/detail/CVE-2022-23491)
Severity: IMPORTANT_VULNERABILITY_SEVERITY
```

As can be easily seen, the version detected as being vulnerable is actually the same as the version that fixes the CVE, but they differ through a leading zero in the minor version number.

**Comment by MaxTranced** [comment]
Version `3.73.0`

As for the repro, I cannot guarantee 100% that this will repro, nor that it will repro forever (as the package may soon get an update), but a repro should be quite trivial IMO: just get a Docker image that installs the `2022.12.7` version of the `certifi` package. Just scan the image generated by this Dockerfile:

```
FROM pytorch/pytorch:1.10.0-cuda11.3-cudnn8-runtime

RUN  set -eux; \
  pip freeze; \
  conda list --name base; \
  conda update --quiet --name base conda --yes; \
  conda clean --all --yes; \
  pip freeze; \
  conda list --name base; \
  test true
```

A quote from the output of the original Dockerfile (the one above is a stripped down version of the original):

```
+ conda update --quiet --name base conda --yes
[redacted]
The following packages will be UPDATED:

  _openmp_mutex                                   4.5-1_gnu --> 5.1-1_gnu
  beautifulsoup4     pkgs/main/noarch::beautifulsoup4-4.10~ --> pkgs/main/linux-64::beautifulsoup4-4.11.1-py37h06a4308_0
  ca-certificates                      2021.9.30-h06a4308_1 --> 2023.01.10-h06a4308_0
  certifi                          2021.10.8-py37h06a4308_0 --> 2022.12.7-py37h06a4308_0
```

**Comment by jvdm** [comment]
Thanks for sharing the details of your Dockerfile, @MaxTranced.

I believe this issue is [fixed in Stackrox Scanner](https://github.com/stackrox/scanner/commit/7ca1e06da7fb34c9bf747c943e67d7f3ddad7964), and since Stackrox Scanner is already pulling from latest commits, I believe this is also fixed in current Stackrox master.

Could you please check if you can reproduce this on the latest `master` branch?

**Comment by MaxTranced** [comment]
Awesome! I will try it when the version with the fix reaches our scanning tools. What version will the fix be included in?

**Comment by jvdm** [comment]
@MaxTranced  The fix should be released officially in 4.0. You are right. The fix is in a different repository. That's the repository of the Stackrox Scanner. That repository holds the vulnerability scanning and feeds used by Stackrox when integrated with Stackrox Scanner.

**Comment by jvdm** [comment]
@MaxTranced I am going to close this. Feel free to re-open or create a new issue if you can reproduce it with the latest master builds or eventually at 4.0. Thanks!
